Jenkinsfile

```diff
@@ -267,33 +270,45 @@ pipeline {
             node('master') {
                 script {
                     deleteDir()
-                    def logServer = [
-                        addr: env.LOG_SERVER,
-                        user: env.LOG_SERVER_USER,
-                        folder: env.LOG_SERVER_FOLDER,
-                        rootDir: env.LOG_ROOT_DIR
-                    ]
-                    def destDir = decideLogsDestination(logServer, env.ZUUL_UUID)
+                    def relLogsDstDir = logsRelPathBasedOnTriggerSource(env.JOB_NAME,
+                        env.BUILD_NUMBER, env.ZUUL_UUID)
 
                     dir('to_publish') {
                         unstash 'processedTestReports'
                         def logFilename = 'log.txt.gz'
-                        obtainLogFile(env.JOB_NAME, env.BUILD_ID, logFilename)
-                        publishToLogServer(logServer, ".", destDir)
-                    }
+                        createCompressedLogFile(env.JOB_NAME, env.BUILD_NUMBER, logFilename)
 
-                    def logsURL = getLogsURL(logServer, env.ZUUL_UUID)
+                        def auth = sshAuthority(env.LOG_SERVER_USER, env.LOG_SERVER)
+                        def dst = logsDirInFilesystem(env.LOG_ROOT_DIR, env.LOG_SERVER_FOLDER, relLogsDstDir)
+                        publishCurrentDirToLogServer(auth, dst)
+                    }
 
+                    def fullLogsURL = logsURL(env.LOG_SERVER, env.LOG_SERVER_FOLDER, relLogsDstDir)
+                    def logDestMsg = "Full logs URL: ${fullLogsURL}"
+                    echo(logDestMsg)
                     if (isGithub()) {
-                        sendGithubComment("Full logs URL: ${logsURL}")
+                        sendGithubComment(logDestMsg)
                     }
+                }
+            }
 
-                    def reportLocationsFile = "${logsURL}/TestReports/WindowsCompute/reports-locations.json"
-                    build job: 'WinContrail/gather-build-stats', wait: false,
-                        parameters: [string(name: 'BRANCH_NAME', value: env.BRANCH_NAME),
-                                     string(name: 'MONITORED_JOB_NAME', value: env.JOB_NAME),
-                                     string(name: 'MONITORED_BUILD_URL', value: env.BUILD_URL),
-                                     string(name: 'TEST_REPORTS_JSON_URL', value: reportLocationsFile)]
+            node('ansible') {
+                script {
+                    deleteDir()
+                    def relLogsDstDir = logsRelPathBasedOnTriggerSource(env.JOB_NAME, env.BUILD_NUMBER, env.ZUUL_UUID)
+                    def fullLogsURL = logsURL(env.LOG_SERVER, env.LOG_SERVER_FOLDER, relLogsDstDir)
+
+                    unstash "Monitoring"
+                    shellCommand('python3', [
+                        'monitoring/collect_and_push_build_stats.py',
```

Comment:
There is logic waiting for status other than `IN_PROGRESS` in the Python script. It is not needed anymore since we are setting status to `SUCCESS` or `FAILURE` here.